### ISSUE TITLE
Problem with source evolution 'Uniform1D'?

### ISSUE THREAD

**Issue by avvliet** [CRPropa/CRPropa3#197]
I received this email from Daniel Maschmann (a master student from Aachen):

'During my master thesis i am studying different source distributions 
with CRPropa3. I am not sure if I am right, but i think that the 
contribution from different distances is not completely right in 
CRPropa3. To explain the problem to you, I composed some slides which 
are attached to this mail. I already presented this in my working group 
but they were not sure about it. Martin Erdmann told me to adress the 
question to you.' 
[source_dist_Crp3_maschmann.pdf](https://github.com/CRPropa/CRPropa3/files/2245606/source_dist_Crp3_maschmann.pdf)

**Comment by rafaelab** [comment]
The analysis seems right, but I don't think this applied to our case. 
The luminosity distance (D_L) corresponds to the distance at which a source with bolometric luminosity L would be if the signal is unattenuated. It accounts for redshift effects during propagation and at emission.
We use the light travel distance (D), related to the lookback time, which corresponds to the time a particle would take to propagate a distance D. If the particle leaves the source and the only energy loss process is adiabatic (due to redshift), I see no reason why this source should have D_L instead of D.  
D_L takes into account the change in wavelength due to the expansion. If one wants to mimick D_L using D instead, the luminosity of the source has to be weighted accodingly, which means that a (1+z) factor has to be included in the injection spectrum.

In summary, the structure of CRPropa is correct in my opinion. If the problem at hand requires D_L to be used, then the luminosity of the sources has to be weighted by (1+z), and therefore CRPropa's internal unit, D, would be equal to D_L.

Does this make sense?

**Comment by TobiasWinchen** [comment]
I think there should be a factor of (1+z)**-1 in the luminosity also for cosmic ray sources as the time intervall between two observed cosmic ray events of the same source gets larger. This cosmic time dilation has been previously included in several Auger anaysis, e.g. also in my thesis.

**Comment by DanielMaschmann** [comment]
Hello, Thanks for your answeres! 
I think my question was not compleetly correct. Of course Crpropa is calculating the light travel distance correctly and this is what we need for the trajectory length. 
My question is more about the implementation of Uniform1D(). Because here i think far away sources are still too luminous.
I set up again some slides with plots where I tried to reproduce the source contribution for the distances. I left it very simple and I hope you will understand this (sorry that i need slides to explaine this but I have no idea how to express this in words...)
[git_hub_discussion_maschmann.pdf](https://github.com/CRPropa/CRPropa3/files/2260260/git_hub_discussion_maschmann.pdf)

**Comment by rafaelab** [comment]
Hi Daniel,
I understand what you are saying. 
I think of it the following way: suppose you have each source emitting one particle, i.e., a uniform luminosity distribution, which is CRPropa's default setting. Then if there are no interactions, in a 1D setup, all particles are detected. What you are saying is that the "right" way to do things is to use the luminosity distance for a uniform source distribution. However, one can just apply an a posteriori 1/(1+z) weight to the CRPropa output to mimick a redshift-dependent evolution and hence the luminosity distance (perhaps one should include the 4pi factor as well). If you try this, I think you'll get what you are expecting. If not, please let us know.
Since CRPropa is supposed to be versatile, I'm not sure  we should implement at function-level (in Uniform1D)  the luminosity scaling as the user may wish to use a specific luminosity evolution and leaving this distribution flat makes it easier for reweighting, in my opinion. 
Therefore, could you please check if adding a 1/(1+z) weight (maybe also the 4pi factor) gets you the desired result? If so, we should include this comment somewhere in the CRPropa wiki so that the users are aware of the underlying physics.

**Comment by DanielMaschmann** [comment]
I applied an a posteriori weight of 1 / (1 + z) to my CrPropa3 output (no 4 * pi factor this comes from the spherical symmetry )  and I got perfectly matching my Monte Carlo-simulation. 
So finally I will just add this factor to my simulations to get the right source evolution.
 
Here is the mc-prediction and the CrPropa output weighted with 1 / (1 + z):
![crpropa_vs_cosmology_source_dist_comoving_distance](https://user-images.githubusercontent.com/42112728/43687993-1e81e818-98e1-11e8-8880-bccd877ddc76.png)

**Comment by rafaelab** [comment]
Thanks for the comparison. 
The 4pi would come in for a uniform 3D source distribution.
If your results match the predictions, then we should  add a note to our wiki to instruct the user about this.

**Comment by TobiasWinchen** [comment]
I think the code and its usage is correct as it is: The equation $$ f = \frac{L}{4\pi r^2(1+z)^2}$$ calculates the observed energy flux emitted by a source with luminosity L in comoving distance r. One of the two factors (1+z) is due to the cosmological time dilaton, the other one due to the change in energy of the particles/photons so that $f = \frac{dE'}{d\Omega dA} = \frac{dE}{d\Omega dA} (1+z)$.
Thus we can define the luminosity distance $r_L = (1+z) r_G$ . However, here we have the number flux of particles and thus only one factor  $(1 + z)$ for the cosmological time dilation. 
See also e.g. 

Berezinsky, V. & Gazizov, A. Z. _Diffusion of Cosmic Rays in the Expanding Universe. I_. Astrophysical Journal, 2006, 643, equation 46, 

respectively surrounding text. We should add a page in the wiki to in detail explain the reasoning and calculations of this (and all other) modules with the plots and appropriate references.

**Comment by TobiasWinchen** [comment]
No modifications to the simulated output data are necessary, the energy loss of the particles is accounted for in the RedShift module, the change of apparent source luminosity due to cosmological time dilation in the SourceUniformRedshift feature.
